### DISCUSSION TITLE
Is the JavaScript component of the BrowserForwarder supposed to be closed-source?

### DISCUSSION THREAD

**Discussion by daichuanwu21** [v2fly/v2ray-core#2629] [General]
Hello! I can't help but notice that the [JS component of the BrowserForwarder](https://github.com/v2fly/v2ray-core/blob/d58649764e3afa52234ada9a28b7ae79577a23a9/release/extra/browserforwarder/index.js) appears to be bundled. This renders its readability similar to that of a binary firmware blob.

Since I'm under the impression that this is an open-source project, my questions are as follows:
1. Is the JS component supposed to be proprietary and therefore distributed as such?
2. If yes, who exactly owns the copyright to it and does it possess any terms of distribution that would conflict with the MIT license of V2Ray? Also, are there any plans to document this for the users?
3. If no, are there any plans to release the component's source code and integrate its build process into V2Ray for better visibility?

Just to clarify, I think that the community has built V2Ray into an awesome piece of software, though I do believe that the inclusion non-free components in an ostensibly free piece of software should be made clear to users.

**Comment by AkinoKaede** [comment]
It's free and open-source.
<https://github.com/v2fly/BrowserBridge>

**Reply by daichuanwu21** [reply]
Ah I see, thanks for clearing up the confusion!
